tensorflow_probability/python/bijectors/gev_cdf.py

```diff
@@ -0,0 +1,178 @@
+# Copyright 2018 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""GEV bijector."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow.compat.v2 as tf
+
+from tensorflow_probability.python.bijectors import bijector
+from tensorflow_probability.python.internal import assert_util
+from tensorflow_probability.python.internal import dtype_util
+from tensorflow_probability.python.internal import tensor_util
+
+
+__all__ = [
+    'GEVCDF',
+]
+
+
+class GEVCDF(bijector.Bijector):
+  """Compute `Y = g(X) = exp(-t(X))`, the GEV CDF,
+  where `t(x) = (1 + shape * (x - loc) / scale) ) ^ (-1 / shape)`.
+
+  This bijector maps inputs from `[-inf, inf]` to `[0, 1]`. The inverse of the
+  bijector applied to a uniform random variable `X ~ U(0, 1)` gives back a
+  random variable with the [Generalized extreme value distribution](
+  https://https://en.wikipedia.org/wiki/Generalized_extreme_value_distribution):
+
+  ```none
+  Y ~ GEVCDF(loc, scale, shape)
+  t(y; loc, scale, shape) = (1 + shape * (y - loc) / scale) ) ^ (-1 / shape)
+  pdf(y; loc, scale, shape) = t(y; loc, scale, shape) ^ (1 + shape) * exp(
+  - t(y; loc, scale, shape) ) / scale
+  ```
+  """
+
+  def __init__(self,
+               loc=0.,
+               scale=1.,
+               shape=0.5,
```

Comment:
Can we change the name of this parameter to something like `concentration`? We generally avoid naming parameters `shape` since it can easily be conflated with `shape` as a TF concept.

Reply:
Sure. That's a better name compared to `shape`.